addons/project/models/project.py

```diff
@@ -1527,14 +1527,30 @@ def _ensure_fields_are_accessible(self, fields, operation='read', check_group_us
                 - False if we are sure the user is a portal user,
         """
         assert operation in ('read', 'write'), 'Invalid operation'
-        if fields and (not check_group_user or self.env.user.has_group('base.group_portal')) and not self.env.su:
+        is_portal = self.env.user.has_group('base.group_portal')
+        if fields and (not check_group_user or is_portal) and not self.env.su:
             unauthorized_fields = set(fields) - (self.SELF_READABLE_FIELDS if operation == 'read' else self.SELF_WRITABLE_FIELDS)
             if unauthorized_fields:
                 raise AccessError(_('You cannot %s %s fields in task.', operation if operation == 'read' else '%s on' % operation, ', '.join(unauthorized_fields)))
+        return is_portal
 
     def read(self, fields=None, load='_classic_read'):
-        self._ensure_fields_are_accessible(fields)
-        return super(Task, self).read(fields=fields, load=load)
+        is_portal = self._ensure_fields_are_accessible(fields)
```

Comment:
Indeed, I agree with you. However, we do that to avoid recall `user_has_groups` or `self.env.user.has_group` to know if the current user is a portal one or not. That's why we return a boolean value in this method. 

I will change it to avoid returning `is_portal` flag in this method.